### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by dchatterjee172**

### CODE REVIEW

File: lorentz.py [hunk 1]
```diff
@@ -148,26 +151,37 @@ def __init__(self, pairwise_matrix, sample_size=10):
...
 
     def __getitem__(self, i):
-        I = torch.Tensor([i]).squeeze().long()
-        while True:
-            j = random.randint(0, self.n_items - 1)
-            if j != i:
+        I = torch.Tensor([i + 1]).squeeze().long()
+        has_child = False
+        arange = np.random.permutation(self.arange)
+        for j in arange:
```
Comment (theSage21) [inline lorentz.py]: pairwise matrix is numpy no? `has_child = (self.pairwise_matrix[i] > 0 ).sum() > 0` will be faster I think?

File: lorentz.py [hunk 2]
```diff
@@ -148,26 +151,37 @@ def __init__(self, pairwise_matrix, sample_size=10):
...
+        arange = np.random.permutation(self.arange)
+        if has_child:
+            indices = [
+                x for x in arange if i != x and self.pairwise_matrix[i, x] < min
+            ][: self.sample_size]
+        else:
+            indices = [
+                x for x in arange if i != x and self.pairwise_matrix[x, i] < min
+            ][: self.sample_size]
         Ks = ([i + 1 for i in [j] + indices] + [0] * self.sample_size)[
```
Comment (theSage21) [inline lorentz.py:181]: j does not need a +1?

Comment (theSage21) [reply lorentz.py:181]: I had thought that's what I did wrong.

**Review by theSage21 — CHANGES_REQUESTED** [review]

**Comment by theSage21** [comment]
just reslove the stuff if it does not matter. I'll merge once that's done

File: lorentz.py (hunk 2 above)
Comment (dchatterjee172) [reply lorentz.py:181]: Yup there's a small mistake. wait!

Comment (dchatterjee172) [reply lorentz.py:181]: `[i + 1 for i in [j] + indices]`
I don't see any problem in this,
as there's 'i + 1` it'll do the j+1

File: lorentz.py (hunk 1 above)
Comment (dchatterjee172) [reply lorentz.py]: yes, changing

**Comment by dchatterjee172** [comment]
Check now!

**Pull request merged**